src/field.jl

```diff
@@ -358,7 +358,7 @@ function tryparsenext(s::StringToken{T}, str, i, len, opts) where {T}
         y2 = iterate(str, i)
     end
 
-    return R(_substring(T, str, i0, i-1)), i
+    return R(_substring(T, str, i0, prevind(str, i))), i
```

Comment:
I think `prevind` is wrong here because `i` is the byte-index of the start of the next character. We want the index up to that point (last byte of previous character) in the substring.

Reply:
Ah, this is interesting. [This code](https://github.com/JuliaComputing/TextParse.jl/blob/68d35e91c9449e422d6a3f85dab6b838046a593f/src/field.jl#L365) requires the byte index of the last character to be included, not the last byte to be included. [This code](https://github.com/JuliaComputing/TextParse.jl/blob/68d35e91c9449e422d6a3f85dab6b838046a593f/src/field.jl#L379) though requires the last byte index of span to be included in the string. Those two things are not the same if the last character takes up more than one codeunit.

The old version of the code had the second case correct. My patch here fixed the first case, but broke the second. I have a new commit coming that fixes both cases :)

Reply:
It does, but for a range end you have to specify the bye index of the last character you want to include :)